### ISSUE TITLE
BIP39 Words Will Not Restore from 4-Letter Abbreviations

### ISSUE THREAD

**Issue by shannona** [BlockchainCommons/GordianSeedTool-iOS#194]
GST will not restore BIP-39 Words from 4-letter abbreviations:
```
fly mule exce reso trea plun nose soda refl adul ramp plan
```
This is all that's stored in some methods such as classic Cryptosteel, so we should be able to restore from it.

**Comment by wolfmcnally** [comment]
Fixed.

**Comment by shannona** [comment]
Tested on a couple of different 4-letter shortened BIP39s, and it restores great.

I did note that if you happen to have something between a 4-letter and a full word it won't restore (e.g., "sprin" instead of "spri" or "spring"), but I think that's fine because there's no reason to store something between 4-letter and full.

**Comment by wolfmcnally** [comment]
Right, that isn't something I'm checking for. You can either use the full word or the first four letters.

**Comment by shannona** [comment]
I just stumbled upon it by accident. Sounds totally reasonable.
